staging/src/k8s.io/client-go/tools/events/helper.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	eventsv1 "k8s.io/api/events/v1"
+	eventsv1beta1 "k8s.io/api/events/v1beta1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+var mapping = map[schema.GroupVersion]string{
+	eventsv1.SchemeGroupVersion:      "regarding",
+	eventsv1beta1.SchemeGroupVersion: "regarding",
+	corev1.SchemeGroupVersion:        "involvedObject",
+}
+
+// GetFieldSelector returns the appropriate field selector based on the API version being used to communicate with the server.
+// The returned field selector can be used with List and Watch to filter desired events.
+func GetFieldSelector(groupVersion schema.GroupVersion, regardingName, regardingKind, regardingUID string) (fields.Selector, error) {
```

Comment:
rename `groupVersion` to `eventsGroupVersion`

add `regardingGroupVersion schema.GroupVersion`, which maps to `involvedObject.apiVersion` or `regarding.apiVersion` if non-empty (kind alone is ambiguous)

Reply:
> rename `groupVersion` to `eventsGroupVersion`
> 
> add `regardingGroupVersion schema.GroupVersion`, which maps to `involvedObject.apiVersion` or `regarding.apiVersion` if non-empty (kind alone is ambiguous)

That seems to be changed into `regardingGroupVersionKind schema.GroupVersionKind`

Reply:
> rename `groupVersion` to `eventsGroupVersion`
> 
> add `regardingGroupVersion schema.GroupVersion`, which maps to `involvedObject.apiVersion` or `regarding.apiVersion` if non-empty (kind alone is ambiguous)

Just like
`func GetFieldSelector(groupVersion schema.GroupVersion, regardingGroupVersionKind schema.GroupVersionKind，regardingName, regardingUID string) (fields.Selector, error)`

Reply:
good call. also consider changing the type of `regardingUID` to `types.UID` (matching the type of the ObjectMeta field), which will prevent accidentally mixing the name and uid args

Reply:
Thank you, I will change this.